starfish/intensity_table.py

```diff
@@ -291,3 +293,123 @@ def mask_small_features(self, size_threshold):
     def _intensities_from_regions(self, props, reduce_op='max') -> "IntensityTable":
         """turn regions back into intensities by reducing over the labeled area"""
         raise NotImplementedError
+
+    def combine_adjacent_features(
+            self, area_threshold, assume_contiguous=True
+    ) -> Tuple["IntensityTable", List, np.ndarray, np.ndarray, np.ndarray]:
+        """
+
+        Parameters
+        ----------
+        area_threshold
+        assume_contiguous
+
+        Notes
+        -----
+
+        --
+        # This is how a non-contiguous ImageStack creation could work:
+        # construct an empty (z, y, x) image
+        decoded_image = np.zeros(self.attrs['image_shape'], dtype=int)
+
+        # fill it with features
+        coordinates = self.indexes['features'].to_frame()[['z', 'y', 'x']]
+        genes = self.gene_name.values
+
+        for i in np.arange(self.sizes[self.Constants.FEATURES.value]):
+            z, y, x = coordinates.iloc[i]
+            decoded_image[z, y, x] = gene_to_int[genes[i]]
+
+        --
+        # ideally we would be able to define a function that would extract the original
+        # area of the now-merged spot and calculate characteristics of it (max intensity, etc)
+        # across the code. It's signature could look a bit like this:
+        def props_to_intensities(props, label_image, image_stack, reduce_function
+                                 ) -> "IntensityTable":
+            tiles: List[np.ndarray] = []
+            for prop in props:
+                # extract the relevant area of the table and apply the reduce function
+                raise NotImplementedError
+
+        Returns
+        -------
+        IntensityTable :
+            spot intensities
+        List[
```

Comment:
forgot to finish all the things this returns?

Reply:
Yikes, sorry.